docs/architecture/adr-017-subspace-fee-grant.md

```diff
@@ -0,0 +1,261 @@
+message MsgRevokeUserAllowance {
+    // the id of the subspace where the granter grants the allowance to the grantee.
+    uint64 subspace_id = 1;
+
+    // the address of the user granting an allowance of their funds.
+    string granter = 2;
+
+    // the address of the user being granted an allowance of another user's funds.
+    string grantee = 3;
+}
+
+// MsgRevokeUserAllowanceResponse defines the Msg/RevokeAllowanceResponse response type.
+message MsgRevokeUserAllowanceResponse {}
+
+// MsgGrantGroupAllowance adds permissions for the group to spend up allowance of fees from the granter inside the given subspace.
+message MsgGrantGroupAllowance {
+    // the id of the subspace where the granter grants the allowance to the grantee.
+    uint64 subspace_id = 1;
+
+    // the id of the group being granted an allowance of another user's funds.  
+    uint32 group_id = 2;
+
+    // the address of the user granting an allowance of their funds.
+    string granter = 3;
+
+    // allowance can be any of fee allowances which implements FeeAllowanceI.
+    google.protobuf.Any allowance = 4 [(cosmos_proto.accepts_interface) = "FeeAllowanceI"];
+}
+
+// MsgGrantGroupAllowanceResponse defines the Msg/GrantAllowanceResponse response type.
+message MsgGrantGroupAllowanceResponse {}
+
+// MsgRevokeGroupAllowance removes any existing allowance from granter to the group inside the subspace.
+message MsgRevokeGroupAllowance {
+    // the id of the subspace where the granter grants the allowance to the group.
+    uint64 subspace_id = 1;
+
+    // the id of the group being granted an allowance of another user's funds.
+    uint32 group_id = 2;
+
+    // the address of the user granting an allowance of their funds.
+    string granter = 3;
+}
+
+// MsgRevokeGroupAllowanceResponse defines the Msg/RevokeAllowanceResponse response type.
+message MsgRevokeUserAllowanceResponse {}
+```
+
+### `Query` Service
+
+In order to allow clients to easily query for allowances we will implement a new queries:
+
+```protobuf
+service Query {
+    // UserAllowances returns all the grants for users.
+    rpc UserAllowances(QueryUserAllowancesRequest) returns (QueryAllowancesResponse) {
+        option (google.api.http).get = "/desmos/subspaces/v3/subspaces/{subspace_id}/granter/{granter}/users/allowances";
+    }
+    // GroupAllowances returns all the grants for groups.
+    rpc GroupAllowances(QueryGroupAllowancesRequest) returns(QueryGroupAllowancesResponse) {
+        option (google.api.http).get = "/desmos/subspaces/v3/subspaces/{subspace_id}/granter/{granter}/groups/allowances";
+    }
+}
+
+// QueryUserAllowancesRequest is the request type for the Query/UserAllowances RPC method.
+message QueryUserAllowancesRequest {
+    // the id of the subspace where the granter grants the allowance to the grantee.
+    uint64 subspace_id = 1;
+
+    // the address of the user granting an allowance of their funds.
```

Comment:
We should make it optional to allow queries all the grants present, and all the grants given a grantee address
```suggestion
    // (optional) the address of the user granting an allowance of their funds.
```